File: HotelBookingAPi/Models/Booking.cs
[Table("Bookings")]
public class Booking
{
    [Key]
    public int BookingId { get; set; }

    public int UserId { get; set; }
    public int RoomId { get; set; }

    public DateTime CheckInDate { get; set; }
    public DateTime CheckOutDate { get; set; }

    public decimal TotalAmount { get; set; }

    public string Status { get; set; } // Confirmed / Cancelled

    // Navigation
    [ForeignKey("UserId")]
    public User User { get; set; }

    [ForeignKey("RoomId")]
    public Room Room { get; set; }
}

File: HotelBooking-UI/src/app/features/booking/booking/booking.ts
export interface Booking {
  bookingId: number;
  userId: number;
  roomId: number;

  checkInDate: Date;
  checkOutDate: Date;

  totalAmount: number;
  status: string;

  // 🔹 Navigation properties (from backend Include)
  room?: Room;
}

export interface Room {
  roomId: number;
  roomNumber: string;
  price: number;

  hotel?: Hotel;
}

export interface Hotel {
  hotelId: number;
  name: string;
  location: string;
}
